board/views.py

```diff
@@ -94,19 +122,45 @@ def post(self, req):
         Fields:
             name -> string: the author of the post, optional.
             message -> string: the message the user wish to convey.
-            photo -> Images: the photos the user uploads.
+            photo -> Image: the photo the user uploads.
         Returns:
-            A response of either status `204` for success or `422` for invalid data.
+            Invalid form data: 400
+            Board UUID not found: 404
+            Success: JSON representation of the post
         """
+
         form = PostForm(req.POST, req.FILES)
 
         if (form.is_valid()):
-            #TODO: save form data to database
-
-            # 204 is an empty response with no content, meaning that the operation was a success
-            return HttpResponse(status=204)
-        # 422 meaning the data is valid but does not match business model
-        return HttpResponse(status=422)
+            try:
+                board_uuid = form.cleaned_data['board']
+                board = Board.objects.get(uuid=board_uuid)
+            except:
+                return HttpResponse(status=404, headers={
+                    'Content-Length': '0',
+                    'Content-Type': 'application/json'
+                })
+            
+            img_file = form.files.get('photo') 
+            if (img_file is None):
+                img = None
+            else:
+                img = Image(name=img_file.name, photo=img_file.file.read())
+                img.save()
+            
+            post = Post(
+                associated_board=board, 
+                name=form.cleaned_data['name'],
+                message=form.cleaned_data['message'],
+                photo=img
+            )
+            post.save()
+            return JsonResponse(get_post_dict(post))
```

Comment:
Will the frontend will be calling this method? If so shouldn't this be returning a redirect, or will the frontend send requests to an entirely different function?

Reply:
what do you mean? The frontend will be sending a POST req to `board:post-create` and the API will handle the rest and give JSON responses.

Reply:
I'm just asking to make sure the frontend properly receives a redirect response from the server when it attempts to add a post